director/schedule_push.go

```diff
@@ -0,0 +1,254 @@
+package director
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/mdmdirector/mdmdirector/db"
+	"github.com/mdmdirector/mdmdirector/types"
+	"github.com/mdmdirector/mdmdirector/utils"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/vmihailenco/taskq/v3"
+)
+
+func ScheduledCheckin(pushQueue taskq.Queue) {
+
+	var task = taskq.RegisterTask(&taskq.TaskOptions{
+		Name: "push",
+		Handler: func(device types.Device) error {
+			err := PushDevice(device)
+			if err != nil {
+				ErrorLogger(LogHolder{Message: err.Error()})
+			}
+			return nil
+		},
+	})
+
+	counter := 0
+	for {
+		if !DevicesFetchedFromMDM {
+			time.Sleep(30 * time.Second)
+			log.Info("Devices are still being fetched from MicroMDM")
+			counter++
+			if counter > 10 {
+				break
+			}
+		} else {
```

Comment:
Do we need an else here? Why not just break?

Reply:
I think we do? We wait for a variable set elsewhere to be true and then break

Reply:
Yeah, what you are saying makes sense. I now understand the logic.